client/store/sagas/sync.ts

```diff
@@ -12,23 +12,24 @@ import {
   deleteSection,
   duplicateItem,
   editItem,
+  setResume,
   setResumeState,
 } from '../resume/resumeSlice';
 
 const DEBOUNCE_WAIT = 1000;
 
-const debouncedSync = debounce((resume: Resume) => updateResume(resume), DEBOUNCE_WAIT);
+const debouncedSync = debounce((resume: Resume, dispatch: AppDispatch) => updateResume(resume).then((resume) => dispatch(setResume(resume))), DEBOUNCE_WAIT);
```

Comment:
@klejejs Just a bit curious about this. Would this not cause side effect changes to the frontend after the fact, for ex. the user types in a bunch of stuff and waits for while, and then continues to type some more details but the promise is already running. This could overwrite the user's current changes, would it not? It's a very very tight window, I'm sure, but more noticeable on slower internet speeds.

What would be the issue if we didn't update state in the sync fn?

Reply:
The issue is that without any database updates, the `lastChanged` date field in DB does not update, thus the PDF generator cannot check if the existing file is up-to-date with data or not.

I could investigate sending only new changes to the server to decrease request size but I believe that could be a lot of work.
Another option would be to detect changes made to UI after the request and not run `dispatch(setResume(resume))` when the request completes if the resume version is newer than the one saved in DB. Of course, then special handling would be needed for cases when the request fails and there should also be some checks to make sure the user does not lose the changes when they suddenly decide to close the page.

What do you think? Which would be the better option?